ui/packages/app/src/project_setting/metadata/Labels.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from "react";
+
+import {
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton
+} from "@elastic/eui";
+import { isDNS1123Label } from "../../validation/validation";
+import config from "../../config";
+
+export const Labels = ({ onChange, initialLabels }) => {
+  const [items, setItems] = useState([]);
+
+  useEffect(() => {
+    if (initialLabels) {
+      setItems(
+        initialLabels.map((label, idx) => ({
+          ...label,
+          idx,
+          isKeyValid: true,
+          isValueValid: true
+        }))
+      );
+    }
+  }, [initialLabels]);
+
+  const addItem = () => {
+    const newItems = [
+      ...items,
+      {
+        idx: items.length,
+        isKeyValid: false,
+        isValueValid: false
+      }
+    ];
+    setItems(newItems);
+    onChange(newItems);
+  };
+
+  const onKeyChange = idx => {
+    return e => {
+      const newItems = [...items];
+      const newKey = e.target.value.trim();
+      newItems[idx] = {
+        ...newItems[idx],
+        key: newKey,
+        isKeyValid: isDNS1123Label(newKey)
+      };
+      setItems(newItems);
+      onChange(newItems);
+    };
+  };
+
+  const onValueChange = idx => {
```

Comment:
this function can be combined with the above just taking the `key` as a input and not using `value` as a hardcoded key in the map